src/components/menu/js/menuServiceProvider.js

```diff
@@ -60,13 +60,19 @@ function MenuProvider($$interimElementProvider) {
         $animate.enter(options.backdrop, $document[0].body);
       }
 
+      // Handles cases where the menu was destroyed, while it was animating.
+      var destroyListener = scope.$on('$destroy', hideBackdrop);
```

Comment:
It is quite unnormal to have a destroy listener inside of the interims `onShow` method.